### ISSUE TITLE
BUG: float32 exp does not output denormals 

### ISSUE THREAD

**Issue by r-devulap** [numpy/numpy#14049]
#### Numpy/Python version information:
Numpy version 1.18.0.dev0+101a8a6
```python
import numpy as np
>>> np.exp(np.float32(-103.00))
0.0
```
Expected output from NumPy 1.16.2: 
```python
import numpy as np
>>> np.exp(np.float32(-103.00))
1e-45
```

**Comment by r-devulap** [comment]
Submitted a PR #14048 to fix this as well. The PR also adds a validation test suite for exp to confirm accuracy.

**Comment by charris** [comment]
Closed by #14048 .
